var data = [{
    "variant": {
        "href": "http://evidence.pgp-hms.org/rs5186",
        "text": "rs5186"
    },
    "clinicalImportance": "Low",
    "impact": "Likely pathogenic\nUnknown, Homozygous",
    "alleleFreq": "21%",
    "summary": "This common noncoding genetic variant has an allele frequency of ~30% and is associated with an increased risk of hypertension. If ~25% of non-carriers have hypertension, Bonnardeaux et al's data predict ~4% increased risk of hypertension per copy of this variant. This SNP is in the 3' noncoding region of the AGTR1 transcript (angiotensin II type 1 receptor), also known as AT2R1 or AT1R, which is a target of hypertension drugs."
},
{
        "variant": {
            "href": "http://evidence.pgp-hms.org/MBL2-R52C",
            "text": "MBL2-R52C"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "4.9%",
        "summary": "This variant is associated with mannose binding protein deficiency which leads to impaired complement system immune response to mannose-rich pathogens. Patients homozygous for this allele or compound heterozygous are likely to have increased susceptibility to infection, but Hellemann et al. report heterosis for intensive care outcomes in heterozygous subjects. The wild-type version of this gene is known as variant allele A, while this is called variant allele D. See G54D (variant B) and G57E (variant C)."
    }]
var data2 = [{
        "variant": {
            "href": "http://evidence.pgp-hms.org/MBL2-R52C",
            "text": "MBL2-R52C"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "4.9%",
        "summary": "This variant is associated with mannose binding protein deficiency which leads to impaired complement system immune response to mannose-rich pathogens. Patients homozygous for this allele or compound heterozygous are likely to have increased susceptibility to infection, but Hellemann et al. report heterosis for intensive care outcomes in heterozygous subjects. The wild-type version of this gene is known as variant allele A, while this is called variant allele D. See G54D (variant B) and G57E (variant C)."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/MTRR-I49M",
            "text": "MTRR-I49M"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "45%",
        "summary": "This common variant (HapMap allele frequency of 31.3%) in a protein involved in folate (B9) and cobalamin (B12) metabolism and is often reported as \"MTRR I22M\" (an alternative transcript position). Mothers homozygous for this variant are associated with having around a increased chance of a child with Down syndrome (risk of 0.4%, average risk in population is 0.25%). Notably, age plays a far larger role in the rate of Down syndrome (risk is 4.5% for a mother 45-years-of-age), and it is unknown how this variant may combine with the effect of age. There are conflicting reports associating this variant with incidence of neural tube defects, possibly when combined with MTHFR A222V."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/MPO-M251T",
            "text": "MPO-M251T"
        },
        "clinicalImportance": "Low",
        "impact": "Uncertain pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "0.91%",
        "summary": "A single sporadic observation implicated this variant in causing myeloperoxidase (MPO) deficiency in a recessive manner, but lacks any statistical significance. MPO deficiency generally causes no symptoms, but there are reports of associations with some cancers."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/CYP21A2-Q319X",
            "text": "CYP21A2-Q319X"
        },
        "clinicalImportance": "High",
        "impact": "Well-established pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "?",
        "summary": "Causes 21-hydroxylase deficiency (congenital adrenal hyperplasia) in a recessive manner, causing sex steroid dysfunction that can alter primary and secondary sexual characteristics. Recorded in ClinVar by OMIM and GeneReviews (http://www.ncbi.nlm.nih.gov/clinvar/RCV000012951/), this variant was first reported by Globerman et al 1988 (PMID: 3267225). Analyzing 200 families, Ezquieta et al 2002 (PMID: 12220458) found this variant had an allele frequency of 8.3% in patients."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/C3-R102G",
            "text": "C3-R102G"
        },
        "clinicalImportance": "Moderate",
        "impact": "Likely pathogenic\nComplex/Other, Heterozygous",
        "alleleFreq": "15%",
        "summary": "This variant (also called C3F) is common in Europeans (10.2% allele frequency), and is associated with age-related macular degeneration. In the US, 1.5% of adults over 40 have the disease, but the incidence increases strongly with age (>15% in women over 80). Assuming an average lifetime risk of ~10%, heterozygous individuals have a ~13% risk and homozygous have ~20%."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/COL4A1-Q1334H",
            "text": "COL4A1-Q1334H"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nDominant, Heterozygous",
        "alleleFreq": "32%",
        "summary": "This common variant has been associated with arterial stiffness and, in Japanese, a small increased risk of myocardial infarction (MI, a.k.a. heart attack). This last observation supported a dominant effect for this variant and, assuming a lifetime risk of 15% for MI, we estimate carriers have an additional risk of 0.5-3%."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/RET-R982C",
            "text": "RET-R982C"
        },
        "clinicalImportance": "Low",
        "impact": "Uncertain pathogenic\nDominant, Heterozygous",
        "alleleFreq": "1.6%",
        "summary": "Initially suspected of causing Hirschsprungâ€™s disease, this rare variant has been later reported as present in unaffected controls (allele frequency around 1%). Supporting a lack of effect: Panini et al. did not find loss of function effects seen in other variants, and Svensson et al. report unaffected relatives in a family who carried the same variant. OMIM lists it as reported to cause increased susceptibility, but there do not appear to be any statistically significant reports supporting this hypothesis."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/KRT5-G138E",
            "text": "KRT5-G138E"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nUnknown, Heterozygous",
        "alleleFreq": "5.2%",
        "summary": "This variant is associated with 1.25x increased risk of basal cell carcinoma (common skin cancer, rarely malignant)."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/HABP2-G534E",
            "text": "HABP2-G534E"
        },
        "clinicalImportance": "Moderate",
        "impact": "Likely pathogenic\nUnknown, Heterozygous",
        "alleleFreq": "2.8%",
        "summary": "Known as the \"Marburg I\" polymorphism, this variant is associated with increased risk of coronary heart disease (CHD) in individuals with elevated cholesterol, triglyceride, and fibrinogen levels (5-fold risk of a CHD event, compared to 2-fold risk increase in people with the same risk factors but do not have this variant). In individuals without these elevated levels, however, carriers of the variant do not appear to have any increased risk of CHD."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/SERPINA1-E288V",
            "text": "SERPINA1-E288V"
        },
        "clinicalImportance": "Low",
        "impact": "Well-established pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "3.0%",
        "summary": "This variant represents the PiS variant in alpha-1-antitrypsin deficiency where a homozygous individual has 60% enzymatic activity. This variant alone is unlikely to much effect, but 3-4% of heterozygotes are compound heterozygous with the more severe PiZ variant, which is associated with an increased risk of emphysema and COPD."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/NOD2-G908R",
            "text": "NOD2-G908R"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nUnknown, Heterozygous",
        "alleleFreq": "1.1%",
        "summary": "This variant is associated with an increased susceptibility to Crohn's disease and other inflammatory bowel diseases (IBD). Assuming an overall prevalence of 0.5% for IBD, this variant implies an increased attributable risk of 2% (5x increased risk, for an overall risk of 2.5% of IBD for people with this variant)."
    }, {
        "variant": {
            "href": "http://evidence.pgp-hms.org/AMPD1-Q12X",
            "text": "AMPD1-Q12X"
        },
        "clinicalImportance": "Low",
        "impact": "Likely pathogenic\nRecessive, Carrier (Heterozygous)",
        "alleleFreq": "9.3%",
        "summary": "Causes Adenosine Deaminase Deficiency in a recessive manner. Most of the time individuals do not report symptoms, but when symptoms do exist they to be post-exercise symptoms of muscle weakness, muscle pain, and getting tired more quickly."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/CETP-A390P",
        "text": "CETP-A390P"
      },
      "clinicalImportance": "Low",
      "impact": "Likely pathogenic\nUnknown, Heterozygous",
      "alleleFreq": "3.9%",
      "summary": "This variant is associated with slightly lower HDL (good) cholesterol, although it has a negligible effect (around 2 mg/dl)."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/AMPD1-Q12X",
        "text": "AMPD1-Q12X"
      },
      "clinicalImportance": "Low",
      "impact": "Likely pathogenic\nRecessive, Homozygous",
      "alleleFreq": "9.3%",
      "summary": "Causes Adenosine Deaminase Deficiency in a recessive manner. Most of the time individuals do not report symptoms, but when symptoms do exist they to be post-exercise symptoms of muscle weakness, muscle pain, and getting tired more quickly."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/PRPH-D141Y",
        "text": "PRPH-D141Y"
      },
      "clinicalImportance": "High",
      "impact": "Uncertain pathogenic\nRecessive, Carrier (Heterozygous)",
      "alleleFreq": "0.37%",
      "summary": "Hypothesized to cause ALS (or increased susceptibility) in a recessive manner, but this is based on a single observation and may lack statistical significance. The mutant protein appears to form abnormal aggregates."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/APC-E1317Q",
        "text": "APC-E1317Q"
      },
      "clinicalImportance": "Moderate",
      "impact": "Uncertain pathogenic\nDominant, Heterozygous",
      "alleleFreq": "0.88%",
      "summary": "This rare variant has been hypothesized to increase risk of colon cancer. Later studies have contradicted this, finding no significant enrichment and concluding the variant does not increase risk."
    },  
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/ATM-S49C",
        "text": "ATM-S49C"
      },
      "clinicalImportance": "Low",
      "impact": "Uncertain pathogenic\nComplex/Other, Heterozygous",
      "alleleFreq": "1.00%",
      "summary": "May be associated with a small increased risk of breast cancer."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/HFE-S65C",
        "text": "HFE-S65C"
      },
      "clinicalImportance": "Low",
      "impact": "Uncertain pathogenic\nRecessive, Carrier (Heterozygous)",
      "alleleFreq": "0.98%",
      "summary": "A \"mild\" variant which may cause a very small increased risk of iron overload (hereditary hemachromatosis)."
    },
    
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/COL9A3-R103W",
        "text": "COL9A3-R103W"
      },
      "clinicalImportance": "Moderate",
      "impact": "Likely pathogenic\nDominant, Heterozygous",
      "alleleFreq": "4.8%",
      "summary": "Carriers of this collagen variant are associated with having a significantly increased risk of lumbar disc disease (~11% total risk compared to a typical risk of 4%)."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/MATN3-T303M",
        "text": "MATN3-T303M"
      },
      "clinicalImportance": "Moderate",
      "impact": "Uncertain pathogenic\nUnknown, Heterozygous",
      "alleleFreq": "1.1%",
      "summary": "An Iceland study implicated this in causing increased osteoarthritis -- in particular, hand osteoarthritis. Because the variant is rare, the statistical significance of observations is weak. The authors estimate that individuals heterozygous for this variant have a 2.1-fold risk for hand osteoarthritis."
    },
    {
      "variant": {
        "href": "http://evidence.pgp-hms.org/APOE-C130R",
        "text": "APOE-C130R"
      },
      "clinicalImportance": "High",
      "impact": "Well-established pathogenic\nComplex/Other, Heterozygous",
      "alleleFreq": "14%",
      "summary": "This is generally known as the ApoE4 allele of ApoE and is associated with increased risk of Alzheimer's. 20-25% of individuals are heterozygous for this variant, and 1-2% are homozygous. Data from Khachaturian et al. suggests an average 7% of all individuals developed Alzheimer's by the age of 80; when this is split by ApoE4 status: 10% of ApoE4 heterozygotes (3% increased attributable risk), 40% of ApoE4 homozygotes (33% increased attributable risk), and 5% of non-carriers (2% decreased attributable risk). Notably, their model suggests 70-75% of people would eventually develop Alzheimer's by the age of 100 regardless of ApoE4 genotype (and 25-30% are resistant, regardless of genotype), but that ApoE4 variants shift the disease onset to occur significantly earlier (4 years earlier for heterozygous carriers, 13 years for homozygotes)."
    }
    // {
    //   "variant": {
    //     "href": "http://evidence.pgp-hms.org/COL9A3-R103W",
    //     "text": "COL9A3-R103W"
    //   },
    //   "clinicalImportance": "Moderate",
    //   "impact": "Likely pathogenic\nDominant, Heterozygous",
    //   "alleleFreq": "4.8%",
    //   "summary": "Carriers of this collagen variant are associated with having a significantly increased risk of lumbar disc disease (~11% total risk compared to a typical risk of 4%)."
    // },
    // {
]
// var svg = d3.select("#svg3454")
// svg.selectAll("circle").data(data2).enter().append("circle").attr({
//     r: 12,
//     cx: function(d, i) {
//         if (i % 2 === 0) {
//             return 415
//         } else {
//             return 322
//         }
//     },
//     cy: function(d, i) {
//         return i * 75 + 74
//     },
//     class: "gene"
// }).style("fill", function(d) {
//     if (d.clinicalImportance == "Low") {
//         return "green"
//     } else if (d.clinicalImportance == "Moderate") {
//         return "yellow"
//     } else if (d.clinicalImportance == "High") {
//         return "red"
//     }
// }).style("opacity", .5).on('mouseover', function(d) {
//     var nodeSelection = d3.select(this).style({
//         opacity: '0.8'
//     });
//     nodeSelection.select("text").style({
//         opacity: '1.0'
//     });
// }).on('mouseout', function(d) {
//     var nodeSelection = d3.select(this).style({
//         opacity: ".5"
//     });
// })
// .on("click", function(d){
//     d3.select(this)
//     .append("text")
//     .attr({
//         x: 100,
//         y: function(d,i){
//             return 100*i + 80
//         }
//     }).text(function(d){
//         console.log(d.summary)
//         return d.summary
//     })
//     .style("fill", "white")
// })

// .on("click", function(d){
//     d3.select(this).append("text").attr({
//         x:100,
//         y:function(d,i){
//             return 80 * i + 80
//         }
//     }).text(function(d){
//         console.log(d.summary)
//         return d.summary
//     })
// })

// svg.append("g").selectAll("g").data(data2).enter().append("rect")
//     .attr("x", 100)
//     .attr("y", 200)
//     .attr("width", 200)
//     .attr("height", 200)
//     .style("fill", "red")
//     .on("click", function(d){
//         d3.select(this).text(function(d){
//             return d.summary
//             console.log("ehllo")
//         }).style("fill", "white")
//         .attr({
//             x:400,
//             y: 400
//         })
        
//     })
var svg = d3.select("#svg3454");
var geneSummary = svg.selectAll("g")
    .data(data2)
    .enter().append("g")
    .attr("transform", function(d, i) { return "translate(100,100)"; });

    geneSummary.append("circle")
    .attr({
    r: 12,
    cx: function(d, i) {
        if (i < 12) {
            return 320
        } else {
            return 220
        }
    },
    cy: function(d, i) {
        if(i>=12){
        return (i-12) * 72 + 65
        }
        else {
        return i * 75 - 170
        }
    },
    class: "gene"
}).style("fill", function(d) {
    if (d.clinicalImportance == "Low") {
        return "#00FF19"
    } else if (d.clinicalImportance == "Moderate") {
        return "yellow"
    } else if (d.clinicalImportance == "High") {
        return "red"
    }
}).style("opacity", .5).on('mouseover', function(d) {
    var nodeSelection = d3.select(this).style({
        opacity: '1'
    }).on('mouseout', function(d){
        var nodeSelection = d3.select(this).style({
            opacity: '.5'
        })
    });
   
}).on("click",function(d){
        console.log(d)
        d3.select(".variantSummary")
            .style("opacity", "1")
            .text(function(){return d.summary})
            .style("color", function() {
    if (d.clinicalImportance == "Low") {
        return "#00FF19"
    } else if (d.clinicalImportance == "Moderate") {
        return "yellow"
    } else if (d.clinicalImportance == "High") {
        return "#FF0006"
    }
    })
})

function geneName (){
    d3.select(".variantSummary")
    .data(data2)
    .enter()
    .text(function(d){
        console.log(d)
        return d.variant.text})
}
function geneSummary (){
    d3.select(".variantSummary")
    .text(function(d){return d.summary})
}

// geneSummary.append("text")
//     .attr("dy", "50px")
//     .attr("dx", "-100px")
//     .attr("opacity", "0")
//     .text(function(d) { return d.summary; })
//     .style({
//         fill: "white",
//         "font-size": "24px",
//         width: "200px"

//     })

// d3.select()




// svg.selectAll("text").data(data2).enter()
// .append("text").text(function(d) {
//     return d.summary
// }).attr({
//     x: 100,
//     y: function(d,i){
//         return 100 * i + 100
//     }
// }).style("fill", "white")
 // d3plus.textwrap()
 //    .container(d3.select("text"))
 //    .resize(true)
 //    .draw();














